### PR TITLE
VideoPlayer: fixes for DemuxClient

### PR DESCRIPTION
**Pull request by FernetMenta**
may fix tvh issues if there were some

jenkins build this please

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp [hunk 1]
```diff
@@ -427,6 +431,13 @@ void CDVDDemuxClient::RequestStreams()
       st->iBlockAlign     = source->iBlockAlign;
       st->iBitRate        = source->iBitRate;
       st->iBitsPerSample  = source->iBitsPerSample;
+      if (source->ExtraSize > 0 && source->ExtraData)
```
Comment (Jalle19) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp]: Maybe move this block to a method?

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp]: I am not a friend of creating a method for 3 lines of code that is only used 2 times in the entire code.

Comment (Jalle19) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp]: Okat

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by fritsch** [comment]
@stefansaraev please give that a test as you were affected as well.

**Comment by stefansaraev** [comment]
yes no more unknown codec id xxx. thanks

**Comment by fritsch** [comment]
Windows again.

**Pull request merged**